### PR TITLE
review eth support

### PR DESCRIPTION
**Pull request by dyzz**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,13 @@ cmake-build-debug/
 /patches/
 /data/
 /src/bitcoin/
+src/CMakeCache.txt
```
Comment (YihaoPeng) [inline .gitignore]: Oh, you run `cmake ..` in `src` directory...

File: CMakeLists.txt [hunk 2]
```diff
@@ -291,7 +291,7 @@ include_directories(src test ${CHAIN_SRC_ROOT}/src ${CHAIN_SRC_ROOT}/src/config
                     ${LIBEVENT_INCLUDE_DIR} ${MYSQL_INCLUDE})
 set(THIRD_LIBRARIES ${BITCOIN_LIBRARIES} ${OPENSSL_CRYPTO_LIBRARY} ${OPENSSL_SSL_LIBRARY} ${Boost_LIBRARIES} ${LIBCONFIGPP_LIBRARY}
                     ${LIBZMQ_LIBRARIES} ${GLOG_LIBRARIES} ${CURL_LIBRARIES} ${ZOOKEEPER_LIBRARIES} ${KAFKA_LIBRARIES} ${LIBEVENT_LIB}
-                    ${LIBEVENT_PTHREADS_LIB} ${secp256k1_LIBRARIES} ${PTHREAD_LIBRARIES} ${MYSQL_LIB} ${GMP_LIBRARIES})
+                    ${LIBEVENT_PTHREADS_LIB} ${secp256k1_LIBRARIES} ${PTHREAD_LIBRARIES} ${MYSQL_LIB} ${GMP_LIBRARIES} /usr/local/lib/libethash.a)
```
Comment (YihaoPeng) [inline CMakeLists.txt]: I will add a `FindLibEthhash.cmake` module to `cmake/Modules`, so `libethash.a` in another directory will be found.

**Comment by YihaoPeng** [comment]
I have merged the pull request to `dev` branch. (https://github.com/btccom/btcpool/pull/64)
Discussion about the pull request is still here but this pull request will not be merged.

File: src/GwMaker.h [hunk 3]
```diff
@@ -36,28 +36,74 @@
...
+  const string userpwd;
+  const string data;
+  const string agent;
+  const string topic;
+  const uint32 interval;
+  shared_ptr<GwHandler> handler;
+  bool enabled;
+};
+
+static vector<GwDefinition> gGwDefiniitons;
```
Comment (YihaoPeng) [inline src/GwMaker.h]: 头文件中惊现文件作用域全局变量:sweat_smile:

Comment (fatrat1117) [reply src/GwMaker.h]: All header files have macro to guarantee that  it is included only time throughout the project.
#ifndef GW_MAKER_H_
#define GW_MAKER_H_
#endif

However, putting definition in .cc file is also a good idea:)

Comment (YihaoPeng) [reply src/GwMaker.h]: 并不是“throughout the project”，而是“throughout the compilation unit”，也就是，如果 a.cc 和 b.cc 各自 include 了这个头文件，它们将各自产生一个自己的 gGwDefiniitons，并且由于 static ，这两个gGwDefiniitons 在链接时对对方并不可见，也不会产生链接错误。然而，结果是产生了不同的 gGwDefiniitons，而并不是统一的全局变量 gGwDefiniitons。

Comment (fatrat1117) [reply src/GwMaker.h]: oh yeah, you are right. move to .cc

**Comment by YihaoPeng** [comment]
We moved to `deveth` branch and it could not be merged into master for a long time.
Bitcoin support is incomplete within the deveth branch.

I will close this pull request. When we are ready for the new version, we will merge it into master.

**Pull request closed without merging**